akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.persistence.scaladsl
+
+import akka.annotation.DoNotInherit
+import akka.annotation.InternalApi
+import akka.typed.Behavior.UntypedBehavior
+import akka.typed.Signal
+import akka.typed.persistence.internal.PersistentActorImpl
+import akka.typed.scaladsl.ActorContext
+
+object PersistentActor {
+  def persistent[Command, Event, State](
+    persistenceId: String,
```

Comment:
this will have to be `ActorContext => String` to be able to work with `ClusterSharding`, since the `persistenceId` is not known when creating the Behavior